lightning/src/blinded_path/message.rs

```diff
@@ -78,16 +79,61 @@ impl Writeable for ReceiveTlvs {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
 		// TODO: write padding
 		encode_tlv_stream!(writer, {
-			(6, self.path_id, option),
+			(65537, self.context, option),
 		});
 		Ok(())
 	}
 }
 
+/// Represents additional data included by the recipient in a [`BlindedPath`].
+///
+/// This data is encrypted by the recipient and remains invisible to anyone else.
+/// It is included in the [BlindedPath], making it accessible again to the recipient
+/// whenever the [BlindedPath] is used.
+/// The recipient can authenticate the message and utilize it for further processing
+/// if needed.
+#[derive(Clone, Debug)]
+pub enum MessageContext {
+	/// Represents the data specific to [`OffersMessage`]
+	///
+	/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+	Offers(OffersContext),
+	/// Represents custom data received in a Custom Onion Message.
+	Custom(Vec<u8>),
+}
+
+/// Contains the data specific to [`OffersMessage`]
+///
+/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+#[derive(Clone, Debug)]
+pub enum OffersContext {
+	/// Represents an unknown BOLT12 payment context.
+	/// This variant is used when we are unsure if we are dealing
+	/// with an Inbound or Outbound BOLT12 payment.
```

Comment:
Hmm... how about:

```rust
/// This variant is used when a message is sent without using a [`BlindedPath`] or over one created prior to LDK version 0.0.124.
```

Reply:
Well, this one's much better! 😄
Updated in [8137b53](https://github.com/lightningdevkit/rust-lightning/commit/8137b53b96d4e7104bde8533b50d622230751ba2)
Thanks for the suggestion!